pkg/sql/colexec/top/top.go

```diff
@@ -158,8 +158,9 @@ func (ctr *container) build(ap *Argument, bat *batch.Batch, proc *process.Proces
 		}
 	}
 	err := ctr.processBatch(ap.Limit, bat, proc)
-	proc.PutBatch(bat)
-	proc.SetInputBatch(batch.EmptyBatch)
+	// WTF is going on?
```

Comment:
This proc.PutBatch and proc.SetInputBatch(empty) is freaking insane.   What the hack does it do?